corehq/apps/app_manager/suite_xml/post_process/resources.py

```diff
@@ -23,18 +23,20 @@ class Meta(object):
 def add_xform_resource_overrides(domain, app_id, pre_to_post_map):
     overrides_by_pre_id = get_xform_resource_overrides(domain, app_id)
     dirty = False
-    for pre, post in pre_to_post_map.items():
-        if pre in overrides_by_pre_id:
-            if post != overrides_by_pre_id[pre].post_id:
-                raise ResourceOverrideError("Cannot change override of {}".format(pre))
+    for pre_id, post_id in pre_to_post_map.items():
+        if pre_id in overrides_by_pre_id:
+            if post_id != overrides_by_pre_id[pre_id].post_id:
+                raise ResourceOverrideError("""
+                    Cannot change override: domain {}, app {}, pre_id {}
+                """.strip().format(domain, app_id, pre_id))
         else:
             dirty = True
             override = ResourceOverride.objects.create(
                 domain=domain,
                 app_id=app_id,
                 root_name=XFormResource.ROOT_NAME,
-                pre_id=pre,
-                post_id=post,
+                pre_id=pre_id,
+                post_id=post_id,
             )
             override.save()
```

Comment:
What happens when this gets saved, but a later value in `pre_to_post_map.items()` raises a `ResourceOverrideError`? Should this be using a transaction over the entire for loop so that this is only committed if there are no errors?

Reply:
8fb2505d205

Reply:
`ResourceOverride.objects.create(...)` still creates the object in the loop. This extra save is basically a no-op. I think what you're looking for is:

```
new_overrides = []
for thing in things:
    if should_override:
        new_overrides.append(ResourceOverride(params))

if new_overrides:
    ResourceOverride.objects.bulk_create(new_overrides)
```

Reply:
thank you - 794b78588f3